haystack/nodes/other/join_docs.py

```diff
@@ -60,7 +68,11 @@ def run_accumulated(self, inputs: List[dict], top_k_join: Optional[int] = None):
         else:
             raise ValueError(f"Invalid join_mode: {self.join_mode}")
 
-        sorted_docs = sorted(scores_map.items(), key=lambda d: d[1], reverse=True)
+        # only sort the docs if that was requested
+        if self.sort_by_score:
+            sorted_docs = sorted(scores_map.items(), key=lambda d: d[1], reverse=True)
```

Comment:
I think this can be changed to support `None` as well, without complicated changes.

Add `from math import inf` at the top, then:
```suggestion
            sorted_docs = sorted(scores_map.items(), key=lambda d: d[1] if d[1] is not None else -inf, reverse=True)
```

We can also raise a warning if `None`s are detected in the list, but at least we won't have crashes and the sorting behavior is quite intuitive imho. What do you think?

Reply:
Note that this change doesn't affect the scores themselves, only the sorting behavior. Scores that were `None` will stay `None`.

Reply:
I am okey to consider all scores with `None` as `-inf` when it comes to sorting - meaning they will be at the end of the sort.
I think that is just as good as placing them to anywhere else in the sorting list.

The only question is whether we would want to inform the user about this or not. Should we raise a warning when there are some `score=None` values and we are asked to sort?

For now I didn't add the warning - so the function is silent about it (which I think is reasonable), but I can add a warning if you think we should.

Reply:
_"Note that this change doesn't affect the scores themselves, only the sorting behavior. Scores that were None will stay None."_ >> Yeap, that is clear, thanks.

Reply:
I though about it a bit and I believe the best here would be an `INFO` level log. I really don't like the code making assumptions silently, but this information is of little relevance in most cases. So INFO sounds like a good compromise. Let's add it and we're done :blush: